src/node.cpp

```diff
@@ -166,34 +167,47 @@ std::vector<session_key_t> Node::getReadySessions() const {
 
 Status Node::demultiplyOutputs(SessionResults& nodeSessionOutputs) {
     auto& [metadata, blobMap] = nodeSessionOutputs.begin()->second;
-    std::vector<NodeSessionMetadata> newSessionMetadatas(metadata.generateSubsessions(getName(), demultiplexCount.value()));
+    SPDLOG_LOGGER_DEBUG(dag_executor_logger, "Will demultiply node: {} outputs to: {} shards", getName(), demultiplexCount.value());
+    auto& tensorDesc = blobMap.begin()->second->getTensorDesc();
+    uint32_t resultsDemultiplyCount = tensorDesc.getDims()[1];
+    std::vector<NodeSessionMetadata> newSessionMetadatas(metadata.generateSubsessions(getName(), resultsDemultiplyCount));
```

Comment:
dims[1] may contain MAX_UINT64 value. This does not fit uint32_t and would overflow. Do we want some kind of limit?

Reply:
Good point